### PR TITLE
Update to use nodejs-sdk v2.0

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -2,63 +2,48 @@
...
+    endpoint: configuration.apiEndpoint,
+    linkResolver: configuration.linkResolver
+  };
+  return prismic.api(configuration.apiEndpoint, {
+    accessToken: configuration.accessToken,
+    req: req
+  });
+}
 
 function handleError(err, req, res) {
```
Comment (dohzya) [inline app.js]: `handleError` is defined only in app.js but called also in blog.js.

I suppose it should be merged with configuration's `onPrismicError`.

File: app.js [hunk 2]
```diff
@@ -2,63 +2,48 @@
...
-
-app.use(errorHandler());
+var prismic = require('prismic-nodejs');
+var app = require('./config');
+var configuration = require('./prismic-configuration');
+var PORT = app.get('port');
+var blog = require('./blog');
+
+// Returns a Promise
+function api(req, res) {
```
Comment (dohzya) [inline app.js]: `api` is defined both in app.js and blog.js, it should be defined only once (for instance in configuration, like onPrismicError).

File: blog.js [hunk 3]
```diff
@@ -1,45 +1,61 @@
...
+  }).then(function(bloghome) {
+    if(bloghome.results[0]) {
       var page = currentPage(req);
       var options = {
         'page' : page,
         'orderings' :'[my.post.date desc]'
       };
-      p.query('[[:d = at(document.type, "post")]]', options, function (err, response) {
-        if (err) { configuration.onPrismicError(err, req, response); return; }
+      api(req, res).then(function(api) {
```
Comment (dohzya) [inline blog.js]: `api()` shouldn't be called more than 1 time during a request.

File: blog.js [hunk 4]
```diff
@@ -1,45 +1,61 @@
...
+        return api.query(prismic.Predicates.at("document.type", "post"), options);
+      }).then(function(response) {
         res.render('bloghome', {
-          'bloghome' : bloghome,
+          'bloghome' : bloghome.results[0],
```
Comment (dohzya) [inline blog.js]: if we used prismic-nodejs v1.2.0, we could use `api.getSingle('post')` which would have returned directly the response result.

**Review by dohzya — COMMENTED** [review]
I needed to use it so I did some modifications myself, feel free to take what you would like: https://github.com/dohzya/nodejs-blog/commit/4317c24d39f2a4e7aa48a2d2c016eca5e346948e 🙂

**Comment by arnaudlewis** [comment]
Pourquoi on update le nodejs-blog ? pour virer express-prismic ? autant passer direct a la derniere version de prismic-nodejs non ?

**Comment by erwan** [comment]
@arnaudlewis please use English on the public repositories!

**Pull request merged**